src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -540,7 +540,8 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
             var inferredPositions = tupleNames.SelectAsArray(n => n != null);
 
             var type = TupleTypeSymbol.Create(syntax.Location,
-                typesBuilder.ToImmutableAndFree(), locationsBuilder.ToImmutableAndFree(),
+                typesBuilder.ToImmutableAndFree().SelectAsArray(TypeMap.AsTypeSymbolWithAnnotations),
```

Comment:
Consider moving the "upgrade" above and changing `typesBuilder` to be `TypeSymbolWithAnnotations` so we don't allocate two arrays. If you think that'll already covered by the PROTOTYPE you added, that's fine too.

Reply:
Done.